pkg/collector/pods_containerlogs_collector.go

```diff
@@ -0,0 +1,148 @@
+		kubeconfig: config,
+	}
+}
+
+func (collector *PODSContainerLogsCollector) GetName() string {
+	return "podscontainerlogs"
+}
+
+// Collect implements the interface method
+func (collector *PODSContainerLogsCollector) Collect() error {
+	containerNamespaces := strings.Fields(os.Getenv("DIAGNOSTIC_CONTAINERLOGS_LIST"))
+
+	// Creates the clientset
+	clientset, err := kubernetes.NewForConfig(collector.kubeconfig)
+	if err != nil {
+		return fmt.Errorf("error in getting access to K8S")
+	}
+	namespace := containerNamespaces[0]
+
+	// List the pods in the given namespace
+	podList, err := getPods(clientset, namespace)
+
+	if err != nil {
+		return err
+	}
+
+	// List all the pods similar to kubectl get pods -n <my namespace>
+	for _, pod := range podList.Items {
+		// Calculate the age of the pod
+		podCreationTime := pod.GetCreationTimestamp()
+		age := time.Since(podCreationTime.Time).Round(time.Second)
+
+		// Get the status of each of the pods
+		podStatus := pod.Status
+
+		var containerRestarts int32
+		var containerReady int
+		var totalContainers int
+
+		// If a pod has multiple containers, get the status from all
+		for container := range pod.Spec.Containers {
+			containerRestarts += podStatus.ContainerStatuses[container].RestartCount
+
+			if podStatus.ContainerStatuses[container].Ready {
+				containerReady++
+			}
+			totalContainers++
+		}
+		containerName := pod.Spec.Containers[0].Name
+		data := fmt.Sprintf("Name: %v, Ready: %v, Status: %v, ReStarts: %v, Ages: %v, ContainerName: %v", pod.Name, fmt.Sprintf("%v/%v", containerReady, containerReady), podStatus.Phase, containerRestarts, age.String(), containerName)
+
+		// Get pods container logs
+		containerLogs, err := getPodContainerLogs(namespace, pod.Name, containerName, clientset)
+
+		if err != nil {
+			return err
+		}
+
+		logData := fmt.Sprintf("%s \n Container Logs for %s: \n%s", data, containerName, containerLogs)
+
+		// Append this to data to be printed in a table
+		collector.data[pod.Name] = logData
+	}
+
+	return nil
+}
+
+func (collector *PODSContainerLogsCollector) GetData() map[string]string {
+	return collector.data
+}
+
+func getPods(clientset *kubernetes.Clientset, namespace string) (*v1.PodList, error) {
+	// Create a pod interface for the given namespace
+	podInterface := clientset.CoreV1().Pods(namespace)
+
+	// List the pods in the given namespace
+	podList, err := podInterface.List(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+	return podList, nil
+}
+
+func getPodContainerLogs(
+	namespace string,
+	podName string,
+	containerName string,
+	clientset *kubernetes.Clientset) (string, error) {
+
+	count := int64(100)
+	podLogOptions := v1.PodLogOptions{
+		Container: containerName,
+		TailLines: &count,
+	}
+
+	podLogRequest := clientset.CoreV1().
+		Pods(namespace).
+		GetLogs(podName, &podLogOptions)
+	stream, err := podLogRequest.Stream(context.Background())
+
+	if err != nil {
+		return "", err
```

Comment:
I'd be nice to have a fmt.Errorf everytime we return an error so that we could easily find where we caught the error in our code

Reply:
Sounds Good, will take care of this in next commit. 👍